internal/provider/dedicated_node_groups_data_source.go

```diff
@@ -0,0 +1,190 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/juju/errors"
+	"github.com/tidbcloud/tidbcloud-cli/pkg/tidbcloud/v1beta1/dedicated"
+)
+
+type dedicatedNodeGroupsDataSourceData struct {
+	ClusterId   types.String     `tfsdk:"cluster_id"`
+	NodeSpecKey types.String     `tfsdk:"node_spec_key"`
```

Comment:
do we need to extract out NodeSpecKey? NodeSpecDisplayName is also same for all items.

Reply:
It's OK, but not necessary. Only two attributes , but the schema will be more complicated